### ISSUE TITLE
Provide example in docs on how to use r.model.setFloatingSpeciesConcentrations

### ISSUE THREAD

**Issue by hsauro** [sys-bio/roadrunner#653]
I couldn't figure out how to use the method:

r.model.setFloatingSpeciesConcentrations

either that or it doesn't work.

r.model.setFloatingSpeciesConcentrations(np.array([1]))
Traceback (most recent call last):

  File "<ipython-input-40-ba48518db726>", line 1, in <module>
    r.model.setFloatingSpeciesConcentrations(np.array([1]))

  File "C:\Tellurium-Winpython-3.7\python-3.7.4.amd64\lib\site-packages\roadrunner\roadrunner.py", line 2191, in setFloatingSpeciesConcentrations
    return _roadrunner.ExecutableModel_setFloatingSpeciesConcentrations(self, *args)

RuntimeError: attempted to access floating species id at index 0

**Comment by evilnose** [comment]
For the model
```
x' = -k1*x
x = 10; k1 = 0.1
```
it seems that roadrunner does not consider "x" to be a floating species but rather a global parameter. Therefore there are no floating species at all.

I think that one would need to make "x" part of a reaction to make it a species? In any case one can also do

```
species x
x' = -k1*x
x = 10; k1 = 0.1
```

The error message is a bit opaque, though. What about "attempted to access floating species id at index 0, but there are 0 floating species"?

**Comment by hsauro** [comment]
We'll have to think about this a bit more. Let's wait until Lucian gets
back.

**Comment by matthiaskoenig** [comment]
The current behavior makes sense, but should be documented.
I.e. if a RateRule is added to a model the changing variable (state
variable) should be a parameter by default.
Only if the user assigns a different object type to the RateRule variable
it should be a different type such as species or compartment.

# parameter rate rule
x' = -k1*x
x = 10; k1 = 0.1

# species rate rule
species x
x' = -k1*x
x = 10; k1 = 0.1

# compartment rate rule
compartment x
x' = -k1*x
x = 10; k1 = 0.1

**Comment by luciansmith** [comment]
I also think we should update the error message.  Something like "attempted to access floating species id at index 0, but there are 0 floating species" might work, but the problem is that thinking in zero-indexed values is not very natural for a biologist:

"Attempted to access floating species at index 1 but there is 1 floating species." 

looks like roadrunner is at fault, and only computer scientists/programmers know otherwise.

Also, the more information about what was specifically being done the better.  So, if we knew the user was setting the value, we should mention this (not sure where the error comes in or how easy it is to modify it in different contexts).

How about this:

"Attempted to access floating species [index#], but there are no floating species in the model."
"Attempted to access floating species [index#], but there is only a single floating species in the model with index '0'."
"Attempted to access floating species [index#], but there are only [#] floating species in the model with indexes '0'-'[#-1]'."

**Comment by luciansmith** [comment]
@matthiaskoenig : is there a place in the documentation that you think should be updated?  I'm not sure where the best point would be.

**Comment by matthiaskoenig** [comment]
@luciansmith This is clearly a model building issue which is part of antimony. So I would document this within the tellurium antimony documention. 
I.e. add a section in
https://tellurium.readthedocs.io/en/latest/antimony.html#examples
which describes RateRules, for instance along the lines of 

---

# RateRules
Rate rules allow to define changes in species, parameters or compartments over time by providing a differential equation expression `x' = dx/dt = f`. To make clear what kind the variable x is its type should be provided within antimony.
 

## Parameter RateRule
By default variables are parameters in an antimony model if not defined otherwise before. To declare a rate rule for the parameter `x` use the `x'` syntax. For instance for the exponential decay of `x`
```
x' = -k1*x
x = 10; k1 = 0.1
```

## Species RateRule
To declare a RateRule for a species make sure the variable `x` is defined as a species
``` 
species x
x' = -k1*x
x = 10; k1 = 0.1
```

## Compartment RateRule
To declare a RateRule for a compartment make sure the variable `x` is defined as a compartment
```
compartment x
x' = -k1*x
x = 10; k1 = 0.1
```

**Comment by evilnose** [comment]
Addressed in tellurium PR [#462](https://github.com/sys-bio/tellurium/pull/462).
